src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -654,7 +654,9 @@ private static void ApplyBindingConfiguration(Collection<FunctionDescriptor> fun
 
         private void HandleHostError(Microsoft.Azure.WebJobs.Extensions.TraceFilter traceFilter)
         {
-            foreach (TraceEvent traceEvent in traceFilter.Events)
+            var events = traceFilter.Events.ToArray();
```

Comment:
Was seeing some "collection modified while enumerating" exceptions, so caching this